File: app/src/main/java/com/bogdan/kolomiiets/birthdayreminder/SettingsActivity.java
package com.bogdan.kolomiiets.birthdayreminder;

import android.content.SharedPreferences;
import android.os.Bundle;
import androidx.appcompat.app.AppCompatActivity;
import android.text.format.DateFormat;
import android.view.View;
import android.widget.Button;
import android.widget.RadioButton;
import android.widget.RadioGroup;
import android.widget.Switch;
import android.widget.TimePicker;

//public class SettingsActivity extends AppCompatActivity implements View.OnClickListener {
//    private Switch switchReminder;
//    private Button btnReminderOK;
//    private RadioGroup radioGroupDay;
//    private RadioButton radioSameDay, radioTheDayBefore, radioTwoDaysBefore;
//    private TimePicker reminderTimePicker;
//    private SharedPreferences preferences;
//
//
//    @Override
//    protected void onCreate(Bundle savedInstanceState) {
//        super.onCreate(savedInstanceState);
//        setContentView(R.layout.activity_settings);
//
//        switchReminder = (Switch) findViewById(R.id.reminder_on_off);
//        btnReminderOK = (Button) findViewById(R.id.ok_btn);
//        btnReminderOK.setOnClickListener(this);
//        radioGroupDay = (RadioGroup) findViewById(R.id.radio_group_container);
//        radioSameDay = (RadioButton) findViewById(R.id.radio_same_day);
//        radioTheDayBefore = (RadioButton) findViewById(R.id.radio_the_day_before);
//        radioTwoDaysBefore = (RadioButton) findViewById(R.id.radio_two_days_before);
//        reminderTimePicker = (TimePicker) findViewById(R.id.reminderTimePicker);
//        reminderTimePicker.setIs24HourView(DateFormat.is24HourFormat(this));
//
//        switchReminder.setOnClickListener(this);
//        preferences = getSharedPreferences(PREFERENCES_NAME, MODE_PRIVATE);
//
//        boolean enableSwitch = preferences.getBoolean(SWITCH, true);
//        if (enableSwitch){
//            switchReminder.setChecked(true);
//            radioTrue();
//            reminderTimePicker.setEnabled(true);
//        } else {
//            switchReminder.setChecked(false);
//            radioFalse();
//            reminderTimePicker.setEnabled(false);
//        }
//
//        //Set active radio button
//        int i = preferences.getInt(CHECKED_RADIO, 0);
//        switch (i){
//            case 0: radioSameDay.setChecked(true);
//            break;
//            case 1: radioTheDayBefore.setChecked(true);
//                break;
//            case 2: radioTwoDaysBefore.setChecked(true);
//                break;
//        }
//
//        //Set text from preferences to etReminderTime
//        reminderTimePicker.setCurrentHour(preferences.getInt(REMINDER_HOUR, 9));
//        reminderTimePicker.setCurrentMinute(preferences.getInt(REMINDER_MINUTE, 0));
//    }
//
//    @Override
//    public void onClick(View v) {
//        switch (v.getId()){
//            case R.id.reminder_on_off:
//                if (switchReminder.isChecked()){
//                    radioTrue();
//                    reminderTimePicker.setEnabled(true);
//                } else {
//                    radioFalse();
//                    reminderTimePicker.setEnabled(false);
//                }
//                break;
//            case R.id.ok_btn:
//                finish();
//        }
//    }
//
//    private void radioTrue(){
//        radioSameDay.setEnabled(true);
//        radioTheDayBefore.setEnabled(true);
//        radioTwoDaysBefore.setEnabled(true);
//    }
//
//    private void radioFalse(){
//        radioSameDay.setEnabled(false);
//        radioTheDayBefore.setEnabled(false);
//        radioTwoDaysBefore.setEnabled(false);
//    }
//
//    @Override
//    protected void onStop() {
//        //Save ic_settings to SharedPreferences
//        preferences = getSharedPreferences(PREFERENCES_NAME, MODE_PRIVATE);
//        SharedPreferences.Editor editor = preferences.edit();
//        boolean checkedSwitch = switchReminder.isChecked() ? true : false;
//        editor.putBoolean(SWITCH, checkedSwitch);
//        switch (radioGroupDay.getCheckedRadioButtonId()) {
//            case R.id.radio_same_day:
//                editor.putInt(CHECKED_RADIO, 0);
//                break;
//            case R.id.radio_the_day_before:
//                editor.putInt(CHECKED_RADIO, 1);
//                break;
//            case R.id.radio_two_days_before:
//                editor.putInt(CHECKED_RADIO, 2);
//                break;
//        }
//        int hour = reminderTimePicker.getCurrentHour();
//        int minute = reminderTimePicker.getCurrentMinute();
//        editor.putInt(REMINDER_HOUR, hour);
//        editor.putInt(REMINDER_MINUTE, minute);
//        editor.commit();
//
//        //Set alarm
//            SetAlarm setAlarm = new SetAlarm();
//            setAlarm.onAlarm(this, checkedSwitch, hour, minute);
//
//        super.onStop();
//    }
//}

File: app/src/main/java/com/bogdan/kolomiiets/birthdayreminder/views/SettingsActivity.kt
package com.bogdan.kolomiiets.birthdayreminder.views

import android.content.Context
import android.content.SharedPreferences
import android.os.Build
import android.os.Bundle
import android.text.format.DateFormat
import android.view.View
import android.widget.Button
import android.widget.RadioGroup
import android.widget.Switch
import android.widget.TimePicker
import androidx.appcompat.app.AppCompatActivity
import androidx.core.view.get
import com.bogdan.kolomiiets.birthdayreminder.Constants
import com.bogdan.kolomiiets.birthdayreminder.Constants.Companion.CHECKED_RADIO
import com.bogdan.kolomiiets.birthdayreminder.Constants.Companion.IS_REMINDER_ON
import com.bogdan.kolomiiets.birthdayreminder.Constants.Companion.RADIO_SAME_DAY
import com.bogdan.kolomiiets.birthdayreminder.Constants.Companion.RADIO_THE_DAY_BEFORE
import com.bogdan.kolomiiets.birthdayreminder.Constants.Companion.RADIO_TWO_DAYS_BEFORE
import com.bogdan.kolomiiets.birthdayreminder.Constants.Companion.REMINDER_HOUR
import com.bogdan.kolomiiets.birthdayreminder.Constants.Companion.REMINDER_MINUTE
import com.bogdan.kolomiiets.birthdayreminder.R
import com.bogdan.kolomiiets.birthdayreminder.utils.setEveryDayAlarm
import com.bogdan.kolomiiets.birthdayreminder.utils.showToast
import java.util.*

class SettingsActivity: AppCompatActivity(), View.OnClickListener {
    private lateinit var reminderOnOff: Switch
    private lateinit var radioGroupContainer: RadioGroup
    private lateinit var reminderTimePicker: TimePicker
    private lateinit var preferences: SharedPreferences
    private lateinit var saveBtn: Button
    private val calendar = Calendar.getInstance()


    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_settings)

        //init UI components
        reminderOnOff = findViewById(R.id.reminder_on_off)
        radioGroupContainer = findViewById(R.id.radio_group_container)
        reminderTimePicker = findViewById(R.id.reminder_time_picker)
        saveBtn = findViewById(R.id.save_btn)

        //set onClickListeners for UI components
        saveBtn.setOnClickListener(this)
        reminderOnOff.setOnCheckedChangeListener { _, isChecked -> enableUiComponents(isChecked) }

        preferences = getSharedPreferences(Constants.PREFERENCES_NAME, Context.MODE_PRIVATE)

        reminderOnOff.isChecked = preferences.getBoolean(IS_REMINDER_ON, false)
        if (!reminderOnOff.isChecked) {
            enableUiComponents(false)
        }
        radioGroupContainer.check(
                when (preferences.getInt(CHECKED_RADIO, RADIO_SAME_DAY)) {
                    RADIO_THE_DAY_BEFORE -> R.id.radio_the_day_before
                    RADIO_TWO_DAYS_BEFORE -> R.id.radio_two_days_before
                    else -> R.id.radio_same_day
                })
        reminderTimePicker.setIs24HourView(DateFormat.is24HourFormat(this))
        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.M) {
            reminderTimePicker.hour = preferences.getInt(REMINDER_HOUR, calendar.get(Calendar.HOUR))
            reminderTimePicker.minute = preferences.getInt(REMINDER_MINUTE, calendar.get(Calendar.MINUTE))
        } else {
            reminderTimePicker.currentHour = preferences.getInt(REMINDER_HOUR, calendar.get(Calendar.HOUR))
            reminderTimePicker.currentMinute = preferences.getInt(REMINDER_MINUTE, calendar.get(Calendar.MINUTE))
        }
    }

    override fun onClick(view: View?) {
        try {
            //Save settings into SharedPreferences
            val editor = preferences.edit()
            editor.putBoolean(IS_REMINDER_ON, reminderOnOff.isChecked)

            editor.putInt(CHECKED_RADIO, when (radioGroupContainer.checkedRadioButtonId) {
                R.id.radio_the_day_before -> RADIO_THE_DAY_BEFORE
                R.id.radio_two_days_before -> RADIO_TWO_DAYS_BEFORE
                else -> RADIO_SAME_DAY
            })

            val tempHour = if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.M)
                reminderTimePicker.hour else reminderTimePicker.currentHour
            editor.putInt(REMINDER_HOUR, tempHour)

            val tempMinute = if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.M)
                reminderTimePicker.minute else reminderTimePicker.currentMinute
            editor.putInt(REMINDER_MINUTE, tempMinute)

            editor.apply()

            setEveryDayAlarm(this, tempHour, tempMinute, reminderOnOff.isChecked)

            getSharedPreferences(Constants.PREFERENCES_EVENTS_NOTIFIER, Context.MODE_PRIVATE)
                    .edit()
                    .clear()
                    .apply()

            showToast(R.string.saved)
            finish()
        } catch (e: Exception) {
            showToast(e.message ?: e.toString())
        }
    }

    private fun enableUiComponents(isChecked: Boolean){
        (0 until radioGroupContainer.childCount).forEach { i ->
            radioGroupContainer[i].isEnabled = isChecked
        }
        reminderTimePicker.isEnabled = isChecked
    }
}